shuffle-coordinator/src/main/java/com/alibaba/flink/shuffle/coordinator/highavailability/zookeeper/ZooKeeperMultiLeaderRetrievalDriver.java

```diff
@@ -155,54 +165,81 @@ public void childEvent(
             CuratorFramework curatorFramework, PathChildrenCacheEvent pathChildrenCacheEvent) {
         String leaderPath = currentLeaderPath.get();
         ChildData childData = pathChildrenCacheEvent.getData();
-        LeaderInformation leaderInfo;
+        boolean initialized = hasInitialized.get();
 
         switch (pathChildrenCacheEvent.getType()) {
             case INITIALIZED:
-                LOG.info("Children cache initialized, begin to retrieve leader information.");
+                List<ChildData> initialData = pathChildrenCacheEvent.getInitialData();
+                LOG.info(
+                        "Children cache initialized, "
+                                + "begin to retrieve leader information: {} - {} - {}.",
+                        initialData == null ? 0 : initialData.size(),
+                        leaderPath,
+                        initialized);
+                hasInitialized.set(true);
+                mayUpdateLeader(initialData);
                 break;
             case CHILD_ADDED:
-                LOG.info("New child node added: {}.", childData.getPath());
+                LOG.info(
+                        "New child node added: {} - {} - {}.",
+                        childData.getPath(),
+                        leaderPath,
+                        initialized);
+                if (!initialized
+                        || (leaderPath != null && !leaderPath.equals(childData.getPath()))) {
+                    return;
+                }
                 mayUpdateLeader(Collections.singletonList(childData));
                 break;
             case CHILD_REMOVED:
-                LOG.info("New child node added: {}.", childData.getPath());
-                if (leaderPath == null || !leaderPath.equals(childData.getPath())) {
+                LOG.info(
```

Comment:
The log `New child node added` in the `CHILD_REMOVED` case branch is a typo? At least it should be different with the log in the `CHILD_ADDED` case branch.

Reply:
Yes, that is a typo. I will fix it.

Reply:
Yes, that is a typo. I will fix it.